doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,183 @@
+
+### Understanding `process.nextTick()`
+
+You may have noticed that `process.nextTick()` was not displayed in the
+diagram, even though its a part of the asynchronous API.  This is because
+`process.nextTick()` is not technically part of the event loop.  Instead, it
+is executed at the end of each phase of the event loop.
+
+Looking back at our diagram, any time you call `process.nextTick()` in a given
+phase, all callbacks passed to `process.nextTick()` will be resolved before
+the event loop continues. This can create some bad situations because **it
+allows you to "starve" your I/O by making recursive `process.nextTick()`
+calls.**  which prevents the event loop from reaching the poll phase.
+
+### Why would that be allowed?
+
+Why would something like this be included in Node.js?  Part of it is a design
+philosophy where an API should always be asynchronous even where it doesn't
+have to be.  Take this code snippet for example:
+
+```js
+function apiCall (arg, callback) {
+  if (typeof arg !== 'string')
+    return process.nextTick(callback,
+      new TypeError('argument should be string'));
+}
+```
+
+The snippet does an argument check and if it's not correct, it will pass the
+error to the callback.  The API updated fairly recently to allow passing
+arguments to `process.nextTick()` allowing it to take any arguments passed
+after the callback to be propagated as the arguments to the callback so you
+don't have to nest functions.
+
+What we're doing is passing an error back to the user.  As far as the _event
+loop_ is concerned, this happens **synchronously**. However, as far as the
+_user_ is concerned, it occurs **asynchronously**: `apiCall()` always runs its
+callback *after* the rest of the user's code.
+
+This philosophy can lead to some potentially problematic situations.  Take
+this snippet for example:
+
+```js
+// this has an asynchronous signature, but calls callback synchronously
+function someAsyncApiCall (callback) { callback(); };
+
+// the callback is called before `someAsyncApiCall` completes.
+someAsyncApiCall(() => {
+
+  // since someAsyncApiCall has completed, bar hasn't been assigned any value
+  console.log('bar', bar); // undefined
+
+});
+
+var bar = 1;
+```
+
+The user defines `someAsyncApiCall()` to have an asynchronous signature,
+actually operates synchronously.  When it is called, the callback provided to
+`someAsyncApiCall ()` is called in the same phase of the event loop because
+`someAsyncApiCall()` doesn't actually do anything asynchronously.  As a
+result, the callback tries to reference `bar` but it may not have that
+variable in scope yet because the script has not been able to run to
+completion.
+
+By placing it in a `process.nextTick()`, the script still has the ability to
+run to completion, allowing all the variables, functions, etc., to be
+initialized prior to the callback being called.   It also has the advantage of
+not allowing the event loop_ to continue.  It may be useful that the user be
```

Comment:
I believe its a typo as I cannot see a reason to italicize "event loop" in this case.